### ISSUE TITLE
Add support for `volumeAttributesClassName` in storage configuration

### ISSUE THREAD

**Issue by scholzj** [strimzi/strimzi-kafka-operator#10462]
`VolumeAttributesClass` introduces to Kubernetes new way how to manage the _performance_ class of the PVCs. Unlike a Storage Class that is immutable, the Volume Attributes Class allows users more flexibility when configuring parameters of the storage such as IOPS, throughput etc. This is something what comes up from time to time from a Strimzi users in our forum. Currently, it is often managed through various annotations. It would be nice to support the configuration of the `VolumeAttributesClass` / `volumeAttributesClassName` in PVCs and in out storage configuration.

**Comment by venkatesh2090** [comment]
Hi, I would like to have a go at this.

Do you reckon this is easy enough for someone new to the code base. I have been able to provision clusters with this feature enabled on AWS, so I am fine with testing this. Couldn't figure it out locally though since the CSI driver for kind doesn't seem to respect this API.

**Comment by scholzj** [comment]
I do not think this would be too complicated. But as it would include some changes to the API, this needs to have a [proposal](https://github.com/strimzi/proposals) first. So you would need to start with that.

**Comment by scholzj** [comment]
I think it is user responsibility to use the `VolumeAttributeClass` that works with given `StorageClass`. So if this happens with a new cluster, I guess Strimzi can just fail and the user has to deal with it. The more interesting questions is what happens if someone changes the `VolumeAttributeClass` for existing volumes. Is it editable property? Or read only?

**Comment by venkatesh2090** [comment]
Looks like you can edit it. But depends on the cloud provider if it will successfully update. e.g. I see this for an aws cluster. Ig we leave that to the user as well.

```
  Warning  VolumeModifyFailed     26s (x5 over 66s)      external-resizer ebs.csi.aws.com                                                          rpc error: code = Internal desc = Could not modify volume "vol-0fa7ff557865862c2": volume "vol-0fa7ff557865862c2" in OPTIMIZING state, cannot currently modify
```

**Comment by venkatesh2090** [comment]
Had a go at this. Would appreciate a review strimzi/proposals#149

**Comment by ppatierno** [comment]
As per discussion on #11210 we decided to wait for the implementation until the feature becomes GA within Kubernetes.

**Comment by andy-townsend** [comment]
Is there any plans to work on this now? It became GA with k8s v1.34 I believe.
